Source/WPEFramework/PluginServer.h

```diff
@@ -2740,7 +2817,18 @@ namespace PluginHost {
                         if (job.IsValid() == true) {
                             Core::ProxyType<Web::Request> baseRequest(Core::proxy_cast<Web::Request>(request));
                             job->Set(Id(), &_parent, service, baseRequest, _security->Token(), !request->ServiceCall());
+#ifdef THUNDER_CRASH_HANDLER
+                            std::string strJsonRpcRequest("");
+                            /* write the json rpc request in new string varaible */
+                            if(!request->ServiceCall() && (baseRequest->HasBody() == true))
+                            {
+                                Core::JSONRPC::Message &message(*(baseRequest->Body<Core::JSONRPC::Message>()));
+                                GetRequestString(message, strJsonRpcRequest);
+                            }
+                            _parent.Submit(Core::proxy_cast<Core::IDispatchType<void>>(job), strJsonRpcRequest);
```

Comment:
???? Oke now I am intersted in the design :-) the workerpool is agnostic (and should be agnostic) to the type of work it is conducting. The scary feeling I get is that the generic job agnostic functionality of the Workerpool is becoming a JSONRPC aware job handler.

Reply:
Digging down, looks like ot is limited to the PluginServer, there is an additional method to submit. A bit lost why it would than not be part of this callstack, but I do think that on R4, we redesign this as I see a lot of code duplication and I   seriously doubt the lifetime management of the IDispatch jobs held in the different lists!